source/parser.hera

```diff
@@ -975,14 +975,27 @@ ThisLiteral
       thisShorthand: true,
     }
   AtThis
-  PrivateThis
+  HashThis
 
-# Adds a #id -> this.#id shorthand as a "PrivateThis" literal in most cases
+# Adds a #id -> this.#id shorthand as a "HashThis" literal in most cases
 # with the exception of keeping `#a in b` as is for branded in checks.
 # https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in#using_the_in_operator_to_implement_branded_checks
 # This is slightly different from how https://262.ecma-international.org/#sec-relational-operators handles relational operators
 # grammatically but should be equivalent in practice
-PrivateThis
+HashThis
+  LengthShorthand:id ( &( _+ ( ( Not __ In ) / In ) ) "" )?:beforeIn ->
```

Comment:
Is there a reason you had to add `""` here?

Reply:
https://github.com/DanielXMoore/Hera/issues/16

Reply:
Ah, I see. Fun. I guess alternatively we could split into two rules, like `PrivateIdentifier`. No strong feeling either way.

Reply:
I generally prefer it as a single rule to prevent backtracking.